config/config.go

```diff
@@ -1,44 +1,49 @@
 package config
 
 import (
-	"encoding/base64"
+	"fmt"
+	"log"
 
-	"github.com/anz-bank/sysl-go/common"
-	"github.com/anz-bank/sysl-go/validator"
-	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
 )
 
-type LibraryConfig struct {
-	Log       LogConfig `yaml:"log"`
-	Profiling bool      `yaml:"profiling"`
-}
+// ReadConfig reads from a single config file and populates both custom, library and genCode config structs
+// cfgFile: path to config file
+// config: a pointer to the custom config struct
+func ReadConfig(cfgFile string, lib *LibraryConfig, gen *GenCodeConfig, config interface{}) {
+	if cfgFile != "" {
+		viper.SetConfigFile(cfgFile)
```

Comment:
Can we do this without `viper` by parsing the yaml contents directly. `viper` is an unnecessary dependency if it can be done without it simply.